llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -5776,6 +5776,18 @@ bool SIInstrInfo::isOperandLegal(const MachineInstr &MI, unsigned OpIdx,
           return false;
       }
     }
+  } else if (isVOP3(MI) && ST.hasNoF16PseudoScalarTransInlineConstants() &&
```

Comment:
For a "slightly" better performance, I'd put `ST.hasNoF16PseudoScalarTransInlineConstants()` before `isVOP3(MI)` because there are definitely more VOP3 instructions than the number of sub targets.

Reply:
Agreed, the subtarget test will be slightly faster than the MI descriptor fetch and bit test.